### ISSUE TITLE
Moving item to bin doesn't update its siblings sort order

### ISSUE THREAD

**Issue by elit0451** [umbraco/Umbraco-CMS#17273]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.5.1, 14.3.0, 15 RC

### Bug summary

When new document nodes are created, they are always added as the last child. But in the case when you delete an item with `sortOrder` 0, and add a new one, it will be added as the first item. It seems like this is caused by not recalculating the `sortOrder` of the siblings when the first item is deleted. This means that the new node comes with `sortOrder = 0`, which results in taking the position as the first item.

### Steps to reproduce

- Use a simple content type and create the following tree structure in the Content section:
```
- Home
  - Child 1
  - Child 2
  - Child 3
```

- Delete "Child 1"
- Add "Child 4"

---
You can also use the content service and verify that the `sortOrder` property of each child doesn't reflect the correct value:
```c#
var parentId = Home.Id;
IEnumerable<IContent> children = ContentService.GetPagedChildren(parentId, 0, int.MaxValue, out _);
```

### Extected:
- "Child 4" to be added as the last item, so the tree looks like this:
```
- Home
  - Child 2
  - Child 3
  - Child 4
```
and the child nodes to have an updated `sortOrder` after deleting "Child 1".

### Actual
- "Child 4" is added as the first item, so the tree looks like:
```
- Home
  - Child 4
  - Child 2
  - Child 3
```

**Comment by elit0451** [comment]
The same issue occurs when Moving the first item and adding a new one

**Comment by manutdkid77** [comment]
Can I pick this one up ? 

Initially we are removing the first item, which has a sortOrder of 0,  and then adding a new item, here it looks like the [SortOrderExists](https://github.com/umbraco/Umbraco-CMS/blob/e4dbf4ce9ff9ca70ed7c82c0c735a4d576aad692/src/Umbraco.Infrastructure/Persistence/Repositories/Implement/ContentRepositoryBase.cs#L1018) method returns false because it cant find any item with sortOrder as 0

**Comment by nul800sebastiaan** [comment]
Fix was reverted, so I am re-opening this issue.

**Comment by nikolajlauridsen** [comment]
Fixed in #17517
